lib/src/errors.dart

```diff
@@ -0,0 +1,87 @@
+// Copyright (c) 2018, the Dart project authors. Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+import 'package:json_annotation/json_annotation.dart';
+import 'package:yaml/yaml.dart';
+
+ParsedYamlException parsedYamlException(String message, YamlNode yamlNode) =>
+    new ParsedYamlException._(message, yamlNode);
```

Comment:
why top level methods instead of a public constructor?

Reply:
Because the type is exported (and nothing else from this library) and I don't want to expose these are constructors.

Reply:
I thought this was exported?

Reply:
Ah nvmd